DNN Platform/Library/Entities/Portals/PortalSettings.cs

```diff
@@ -699,15 +699,19 @@ public string GetProperty(string propertyName, string format, CultureInfo format
             var isPublic = true;
             switch (lowerPropertyName)
             {
+                case "scheme":
+                    propertyNotFound = false;
+                    result = SSLEnabled ? "https" : "http";
+                    break;
                 case "url":
                     propertyNotFound = false;
                     result = PropertyAccess.FormatString(PortalAlias.HTTPAlias, format);
                     break;
-                case "fullurl": //return portal alias with protocol
+                case "fullurl": //return portal alias with protocol - note this depends on HttpContext

```

Comment:
Does `PortalAlias.HTTPAlias` depend on `HttpContext`?  I think if you use the `PortalSettings` constructor that takes a `PortalAliasInfo`, it shouldn't depend on context.

Reply:
@bdukes : If you follow the AddHttp method, it uses HttpContext. I spent some time tracing this code and thought it was a good idea to include this in the comments as it means it is not safe for background tasks.

Reply:
`Globals.AddHTTP` does reference `HttpContent.Current`, but it checks for `null` first, so it should be safe to use from a scheduled task.